src/main/java/org/javarosa/core/util/DataUtil.java

```diff
@@ -64,7 +64,43 @@ public static String[] splitOnSpaces(String s) {
         if ("".equals(s)) {
             return new String[0];
         }
-        return s.split("[ ]+");
+
+        Vector<String> items = new Vector<String>();
```

Comment:
I think this could be replaced with:
```java
reader = CsvReader.parse(s)
reader.readRecord()
String[] values = reader.getValues();
```

Reply:
I definitely wrote the javascript version of this first and just copied it here and tweaked it until it compiled. That's much better.

One thing I'm stuck on - `CsvReader` is in a different package(?), it's under `src/translate` instead of `src/main`. I'm having trouble importing it correctly, running into errors with the package not existing. Do I need to modify gradle to make the CsvReader availabl to this code?

Reply:
ah yes, that does make it harder. The `CsvReader` class is also surprisingly large. I think let's stick with this for now.